src/all/mitaku/src/eu/kanade/tachiyomi/extension/all/mitaku/Mitaku.kt

```diff
@@ -0,0 +1,187 @@
+
+    override fun mangaDetailsParse(document: Document) = SManga.create().apply {
+        status = SManga.COMPLETED
+        update_strategy = UpdateStrategy.ONLY_FETCH_ONCE
+        with(document.selectFirst("article")!!) {
+            title = selectFirst("h1")!!.text()
+            val catGenres = select("span.cat-links a").joinToString { it.text() }
+            val tagGenres = select("span.tag-links a").joinToString { it.text() }
+            genre = listOf(catGenres, tagGenres).filter { it.isNotEmpty() }.joinToString()
+        }
+    }
+
+    // ============================== Chapters ==============================
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
+        val chapter = SChapter.create().apply {
+            url = manga.url
+            chapter_number = 1F
+            name = "Chapter"
+        }
+
+        return Observable.just(listOf(chapter))
+    }
+
+    override fun chapterListSelector(): String {
+        throw UnsupportedOperationException()
+    }
+
+    override fun chapterFromElement(element: Element): SChapter {
+        throw UnsupportedOperationException()
+    }
+
+    // =============================== Pages ================================
+
+    override fun pageListParse(document: Document): List<Page> {
+        val imageElements = document.select("a.msacwl-img-link")
+
+        return imageElements.mapIndexed { index, element ->
+            val imageUrl = element.absUrl("data-mfp-src")
+            Page(index, imageUrl = imageUrl)
+        }
+    }
+
+    override fun imageUrlParse(document: Document): String {
+        throw UnsupportedOperationException()
+    }
+
+    open class UriPartFilter(
+        displayName: String,
+        private val valuePair: Array<Pair<String, String>>,
+    ) : Filter.Select<String>(displayName, valuePair.map { it.first }.toTypedArray()) {
+        fun toUriPart() = valuePair[state].second
+    }
+
+    class CategoryFilter : UriPartFilter(
+        "Category",
+        arrayOf(
+            Pair("Any", ""),
+            Pair("Ero Cosplay", "/ero-cosplay"),
+            Pair("Nude", "/nude"),
+            Pair("Sexy Set", "/sexy-set"),
+            Pair("Online Video", "/online-video"),
+        ),
+    )
+    override fun getFilterList(): FilterList = FilterList(
+        Filter.Header("NOTE: Only one tag search"),
+        Filter.Separator(),
+        CategoryFilter(),
+        TagFilter(),
+    )
+
+    class TagFilter : Filter.Text("Tag") {
+        fun toUriPart(): String {
+            // Replace spaces with hyphens
+            val tag = state.replace(" ", "-")
+            // Encode special characters
+            val url = HttpUrl.Builder()
+                .scheme("https")
+                .host("example.com") // Dummy host to use HttpUrl.Builder
+                .addPathSegment(tag)
+                .build()
+            return url.encodedPath.substring(1) // Remove the leading slash
```

Comment:
Seems needlessly complicated. When you return the unencoded tag and use it in `.addPathSegment()` it will encode it then. If you want to keep the implementation in this function (don't), you would need to use `.addEncodedPathSegment()` to not double encode it.